spring-cloud-stream/src/test/java/org/springframework/cloud/stream/function/FunctionSupportTests.java

```diff
@@ -0,0 +1,121 @@
+import org.springframework.cloud.stream.binder.test.OutputDestination;
+import org.springframework.cloud.stream.binder.test.TestChannelBinderConfiguration;
+import org.springframework.cloud.stream.messaging.Source;
+import org.springframework.context.ApplicationContext;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Import;
+import org.springframework.integration.dsl.IntegrationFlow;
+import org.springframework.integration.dsl.IntegrationFlowBuilder;
+import org.springframework.integration.dsl.IntegrationFlows;
+import org.springframework.integration.scheduling.PollerMetadata;
+import org.springframework.integration.support.MessageBuilder;
+import org.springframework.messaging.Message;
+import org.springframework.messaging.MessageHeaders;
+import org.springframework.scheduling.support.PeriodicTrigger;
+import org.springframework.util.MimeTypeUtils;
+import org.springframework.util.StringUtils;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+/**
+ *
+ * @author Oleg Zhurakousky
+ *
+ * @since 2.1
+ *
+ */
+public class FunctionSupportTests {
+
+	@Test
+	public void testFunctionIsAppliedToMessageSource() {
+		ApplicationContext context = new SpringApplicationBuilder(TestChannelBinderConfiguration.getCompleteConfiguration(MyFunctionsConfiguration.class))
+				.web(WebApplicationType.NONE).run("--spring.cloud.stream.function.name=toUpperCase",  "--spring.jmx.enabled=false");
+
+		OutputDestination target = context.getBean(OutputDestination.class);
+		assertThat(target.receive(1000).getPayload()).isEqualTo("HELLO FUNCTION".getBytes(StandardCharsets.UTF_8));
+	}
+
+	@Test
+	public void testComposedFunctionIsAppliedToMessageSource() {
+		ApplicationContext context = new SpringApplicationBuilder(TestChannelBinderConfiguration.getCompleteConfiguration(MyFunctionsConfiguration.class))
+				.web(WebApplicationType.NONE).run("--spring.cloud.stream.function.name=toUpperCase|concatWithSelf", "--spring.jmx.enabled=false");
+
+		OutputDestination target = context.getBean(OutputDestination.class);
+		assertThat(target.receive(1000).getPayload()).isEqualTo("HELLO FUNCTION:HELLO FUNCTION".getBytes(StandardCharsets.UTF_8));
+	}
+
+	@SpringBootApplication
+	@Import(SourceConfiguration.class)
+	public static class MyFunctionsConfiguration {
+		@Bean
+		public Function<String, String> toUpperCase() {
+			return x -> x.toUpperCase();
+		}
+
+		@Bean
+		public Function<String, String> concatWithSelf() {
+			return x -> x + ":" + x;
+		}
+	}
+
+	@EnableBinding(Source.class)
+	public static class SourceConfiguration {
+
+		@Autowired
+		private Source source;
+
+		@Autowired
+		private FunctionProperties functionProperties;
+
+		@Bean
+		public IntegrationFlow messageSourceFlow(FunctionSupport<?, ?> functionSupport) {
+			Supplier<Message<String>> messageSource = () -> MessageBuilder.withPayload("hello function")
+					.setHeader(MessageHeaders.CONTENT_TYPE, MimeTypeUtils.TEXT_PLAIN).build();
```

Comment:
if you mean 'assert' headers when I receive a message, then yeah you are correct, hence initial. The primary point for this PR is to validate the wiring and invocation of a function(s).
As it progresses it will be added, but I'll address some of it as a PR change request.